RequestService/RequestService.Core/Services/ChartDataService.cs

```diff
@@ -0,0 +1,179 @@
+
+            var dataItems = await _repository.GetActivitiesByMonth(groupId, minDate);
+
+            //Get distinct activities from list
+            var supportActivities = dataItems.Select(x => x.Series).Distinct().ToList();
+
+            List<DataPoint> dataPoints = new List<DataPoint>();
+
+            //Populate chartitems with support activities for each month
+            while (minDate < DateTime.UtcNow.Date)
+            {
+                supportActivities.ForEach(sa =>
+                {
+                    dataPoints.Add(new DataPoint() { Value = 0, XAxis = $"{minDate:yyyy}-{minDate:MM}", Series = sa });
+                });
+                minDate = minDate.AddMonths(1);
+            }
+            
+            GroupAndReplaceValuesForKnown(dataPoints, dataItems);
+            return dataPoints;
+        }
+
+        public async Task<List<DataPoint>> RecentActiveVolunteersByVolumeAcceptedRequests(int groupId)
+        {
+            DateTime minDate = DateTime.UtcNow.Date.AddMonths(-13);
+            var dataItems = await _repository.RecentActiveVolunteersByVolumeAcceptedRequests(groupId, minDate);
+
+            Dictionary<string, (int minValue, int maxValue)> dictCategories = new Dictionary<string, (int minValue, int maxValue)>();
+            dictCategories.Add("1 accepted request", (1, 1));
+            dictCategories.Add("2-3 accepted requests", (2, 3));
+            dictCategories.Add("4-5 accepted requests", (4, 5));
+            dictCategories.Add("6-9 accepted requests", (6, 9));
+            dictCategories.Add("10-more accepted requests", (10, int.MaxValue));
+
+            List<DataPoint> dataPoints = new List<DataPoint>();
+
+            var groupedByUser = dataItems.GroupBy(x => x.Value)
+                        .Select(s => new
+                        {
+                           UserID = s.Key,
+                           Count = s.Count()
+                        });
+
+            foreach (var item in dictCategories)
+            {
+                dataPoints.Add(new DataPoint()
+                {
+                    XAxis = item.Key,
+                    Value = groupedByUser.Count(x=> x.Count >= item.Value.minValue && x.Count <= item.Value.maxValue),
+                    Series = "Dataset 1"
+                });
+            }
+
+            return dataPoints;
+        }
+
+        public async Task<List<DataPoint>> RequestVolumeByActivity(int groupId)
+        {
+            DateTime minDate = DateTime.UtcNow.Date.AddMonths(-13);
+            var dataItems = await _repository.RequestVolumeByActivity(groupId, minDate);
+
+            //Get distinct activities from list
+            var supportActivities = dataItems.Select(x => x.Series).Distinct().ToList();
+
+            List<DataPoint> dataPoints = new List<DataPoint>();
+
+            //Populate chartitems with support activities for each month
+            while (minDate < DateTime.UtcNow.Date)
```

Comment:
This code is repeated above.  Candidate for private method?

Reply:
Also similar code below with jobStatuses.  Maybe we could have a private method that takes a list of strings and a datetime, and returns a List<DataPoint>